File: moneylog_spring/src/main/java/bitc/fullstack503/moneylog_spring/dto/SearchDTO.java
package bitc.fullstack503.moneylog_spring.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;

@Data
public class SearchDTO {
  @JsonProperty("type")
  private String type;  // "income" 또는 "expense" 구분
  @JsonProperty("seq")
  private int seq;
  @JsonProperty("category")
  private String category;
  @JsonProperty("source")
  private String source;
  @JsonProperty("money")
  private int money;
  @JsonProperty("use")
  private String use;
  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
  private Date date;
  @JsonProperty("memberId")
  private String memberId;
}

File: moneylog_app/app/src/main/java/bitc/example/app/dto/SearchDTO.kt
package bitc.example.app.dto

import com.google.gson.annotations.SerializedName

data class SearchDTO (
  @SerializedName("type")
  val type: String,         // 수입(income) 또는 지출(expense)
  @SerializedName("seq")
  val seq: Int,              // 거래 ID
  @SerializedName("category")
  val category: String,     // 카테고리
  @SerializedName("source")
  val source: String,     // 결제 수단
  @SerializedName("money")
  val money: Int,          // 금액
  @SerializedName("date")
  val date: String, // 거래 날짜
  @SerializedName("use")
  val use: String,// 내역
  @SerializedName("memberId")
  val memberId:String
)